packages/engine/src/worker/runner/javascript/state.js

```diff
@@ -88,6 +91,7 @@
       this.__cols = group_state.__agent_batch.cols;
       this.__msgs = group_state.__msg_batch.cols.messages;
       this.__idx_in_group = i_agent_in_group;
+      this.__i_behavior = undefined;
```

Comment:
Is this still needed as we return `this[BEHAVIOR_INDEX_FIELD_KEY]` in `behaviorIndex()`